assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -10310,35 +10332,56 @@ public void doView_grade(RunData data)
 
 		ParameterParser params = data.getParameters();
 
-		state.setAttribute(VIEW_GRADE_SUBMISSION_ID, params.getString("submissionId"));
-		
+		String submissionReference = params.getString("submissionId");
+
+		prepareStudentViewGrade(state, submissionReference);
+	} // doView_grade
+
+	/**
+	 * Prepares the state for the student to view their grade
+	 */
+	private void prepareStudentViewGrade(SessionState state, String submissionReference)
+	{
+		state.setAttribute(VIEW_GRADE_SUBMISSION_ID, submissionReference);
+
 		String _mode = MODE_STUDENT_VIEW_GRADE;
 
-		AssignmentSubmission _s = getSubmission((String) state.getAttribute(VIEW_GRADE_SUBMISSION_ID), "doView_grade", state );
+		AssignmentSubmission _s = getSubmission((String) state.getAttribute(VIEW_GRADE_SUBMISSION_ID), "doView_grade", state);
 		// whether the user can access the Submission object
-		if (_s != null) {
-		    // show submission view unless group submission with group error
-		    Assignment a = _s.getAssignment();
-		    User u = (User) state.getAttribute(STATE_USER);
-		    if (a.isGroup()) {
-		        Collection groups = null;
-		        Site st = null;
-		        try {
-		            st = SiteService.getSite((String) state.getAttribute(STATE_CONTEXT_STRING));
-		            groups = getGroupsWithUser(u.getId(), a, st);
-		            Collection<String> _dupUsers = checkForGroupsInMultipleGroups(a, groups, state, rb.getString("group.user.multiple.warning"));
-		            if (_dupUsers.size() > 0) {
-		                _mode = MODE_STUDENT_VIEW_GROUP_ERROR;
-		                state.setAttribute(VIEW_SUBMISSION_ASSIGNMENT_REFERENCE, _s.getAssignmentId());
-		            }
-		        } catch (IdUnusedException iue) {
-		            M_log.warn(this + ":doView_grade found!" + iue.getMessage());
-		        }
-		    }
+		if (_s != null)
+		{
+			String status = _s.getStatus();
+			if ("Not Started".equals(status))
+			{
+				addAlert(state, rb.getString("stuviewsubm.theclodat"));
```

Comment:
The following code always needs to be executed. The alert is just to warn students that the close date has passed (there's no indication otherwise that they missed the deadline).